src/HttpClientInterception/DelegateHelpers.cs

```diff
@@ -110,4 +115,100 @@ internal static class DelegateHelpers
 
         return async (message, _) => await onIntercepted(message).ConfigureAwait(false);
     }
+
+    /// <summary>
+    /// Converts a collection of function delegates for a set of predicates to match a request
+    /// <see cref="Task{TResult}"/> which returns <see langword="true"/>.
+    /// </summary>
+    /// <param name="predicates">A collection of delegates to convert.</param>
+    /// <returns>
+    /// The converted delegate if <paramref name="predicates"/> has a value; otherwise an exception is thrown if <paramref name="predicates"/> is null.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="predicates"/> is <see langword="null"/>.
+    /// </exception>
+    /// <exception cref="InvalidOperationException">
+    /// If any <paramref name="predicates"/> value is <see langword="null"/>.
+    /// </exception>
+    internal static Func<HttpRequestMessage, Task<bool>> ConvertToBooleanTask(Func<HttpRequestMessage, Task<bool>>[] predicates)
+    {
+        if (predicates == null)
+        {
+            throw new ArgumentNullException(nameof(predicates));
+        }
+
+        foreach (Func<HttpRequestMessage, Task<bool>> predicate in predicates)
+        {
+            if (predicate == null)
+            {
+                throw new InvalidOperationException("At least one predicate is null");
+            }
+        }
+
+        if (predicates.Length == 1)
+        {
+            return predicates[0];
+        }
+
+        return async (message) =>
+        {
+            foreach (Func<HttpRequestMessage, Task<bool>> predicate in predicates)
+            {
+                if (!await predicate(message).ConfigureAwait(false))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        };
+    }
+
+    /// <summary>
+    /// Converts a collection of function delegates for a set of predicates to match a request
+    /// <see cref="Task{TResult}"/> which returns <see langword="true"/>.
+    /// </summary>
+    /// <param name="predicates">A collection of delegates to convert.</param>
+    /// <returns>
+    /// The converted delegate if <paramref name="predicates"/> has a value; otherwise an exception is thrown if <paramref name="predicates"/> is null.
+    /// </returns>
+    /// <exception cref="ArgumentNullException">
+    /// <paramref name="predicates"/> is <see langword="null"/>.
+    /// </exception>
+    /// <exception cref="InvalidOperationException">
+    /// If any <paramref name="predicates"/> value is <see langword="null"/>.
+    /// </exception>
+    internal static Func<HttpRequestMessage, Task<bool>> ConvertToBooleanTask(Predicate<HttpRequestMessage>[] predicates)
```

Comment:
These two functions are so similar that there's a desire to extract the 'common' bits. However, I couldn't manage to do it in a way that made sense. I'm open to suggestions.

Reply:
@martincostello, do you have any options? ☝️

Reply:
I have neither options or opinions.